File: D01/solve.go
package main

import (
	"fmt"
)

func main() {
	data := make([]string, 0)
	digit_words := []string{"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}
	for {
		var input string
		n, _ := fmt.Scanln(&input)
		// if err != nil {
		// 	panic("An error occured")
		// }
		if n == 0 {
			break
		}
		data = append(data, input)
	}
	output := make([]int, 0)

	for _, code := range data {
		c := 0
	out:
		for ch := 0; ch < len(code); ch++ {
			char := code[ch]
			if char >= '0' && char <= '9' {
				c += int(char - '0')
				break
			}
			for index, digit_word := range digit_words {
				if code[ch:min(ch+len(digit_word), len(code))] == digit_word {
					c += index
					break out
				}
			}
		}
		c *= 10
	out2:
		for ch := len(code) - 1; ch >= 0; ch-- {
			char := code[ch]
			if char >= '0' && char <= '9' {
				c += int(char - '0')
				break
			}
			for index, digit_word := range digit_words {
				if code[ch:min(ch+len(digit_word), len(code))] == digit_word {
					c += index
					break out2
				}
			}

		}
		output = append(output, c)
	}

	sum := 0
	for _, o := range output {
		sum += o
	}

	fmt.Println(sum)
}

File: D01/solve.py
import re

data = []

while True:
    new = input()
    if new == "":
        break
    data.append(new)

REGEX_P1 = re.compile(r"[0-9]") # Chall 1
REGEX = re.compile(r"(?=([0-9]|one|two|three|four|five|six|seven|eight|nine))") # Chall 2
MAPPING = {"one": 1, "two": 2, "three": 3, "four": 4, "five":5, "six": 6, "seven": 7, "eight": 8, "nine": 9, "1": 1, "2": 2, "3": 3, "4": 4, "5": 5, "6": 6, "7": 7, "8": 8, "9": 9}

result = sum(10*MAPPING[REGEX_P1.findall(line)[0]] + MAPPING[REGEX_P1.findall(line)[-1]] for line in data) # Chall 1

print("Part 1:", result)

result = sum(10*MAPPING[REGEX.findall(line)[0]] + MAPPING[REGEX.findall(line)[-1]] for line in data) # Chall 1

print("Part 2:", result)